templates/components/multiagent/python/app/workflows/tools.py

```
@@ -0,0 +1,247 @@
+            ctx,
+            tool,
+            tool_call,
+            event_emitter=lambda msg: ctx.write_event_to_stream(
+                AgentRunEvent(
+                    name=agent_name,
+                    msg=msg,
+                    event_type=AgentRunEventType.PROGRESS,
+                    data={
+                        "id": progress_id,
+                        "total": total_steps,
+                        "current": i,
```

Ruff

Function definition does not bind loop variable `i`

(B023)